projects/flimnap/main.cc

```diff
@@ -97,11 +116,15 @@ namespace {
         return next;
     }
 
-    dacsample_t set_handlebar_velocity(float velocity) {
-        // limit velocity to a maximum magnitude of 100 deg/s
-        // input is in units of rad/s
-        const float saturated_velocity = util::clamp(velocity, -sa::MAX_KOLLMORGEN_VELOCITY, sa::MAX_KOLLMORGEN_VELOCITY);
-        const dacsample_t aout = saturated_velocity/sa::MAX_KOLLMORGEN_VELOCITY*sa::DAC_HALF_RANGE + sa::DAC_HALF_RANGE;
+    dacsample_t set_handlebar_reference(float reference) {
+#if defined(USE_BICYCLE_KINEMATIC_MODEL) || defined(USE_BICYCLE_AREND_MODEL)
+        constexpr float MAX_REF_VALUE = sa::MAX_KOLLMORGEN_TORQUE;
+#else
+        // limit velocity to a maximum magnitude of 100 deg/s, input units of rad/s
+        constexpr float MAX_REF_VALUE = sa::MAX_KOLLMORGEN_VELOCITY;
+#endif
+        const float saturated_reference = util::clamp(reference, -MAX_REF_VALUE, MAX_REF_VALUE);
+        const dacsample_t aout = saturated_reference/MAX_REF_VALUE*sa::DAC_HALF_RANGE + sa::DAC_HALF_RANGE;
```

Comment:
Yeah. `MAX_REF_VALUE` is expressed in whatever SI unit. `DAC_HALF_RANGE` is in terms of `daccount_t`.

I can replace the `#if` with a check on the observer type.